packages/react/src/ReactServerContext.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {REACT_PROVIDER_TYPE, REACT_CONTEXT_TYPE} from 'shared/ReactSymbols';
+
+import type {
+  ReactServerContext,
+  ServerContextJSONValue,
+} from 'shared/ReactTypes';
+
+const globalRegistry: {
+  [globalName: string]: ReactServerContext<any>,
+} = {};
+
+export function createServerContext<T: ServerContextJSONValue>(
+  globalName: string,
+  defaultValue: T,
+): ReactServerContext<T> {
+  if (globalRegistry[globalName]) {
+    throw new Error('ServerContext in that name already exists');
+  }
+  const context: ReactServerContext<T> = {
+    $$typeof: REACT_CONTEXT_TYPE,
```

Comment:
I haven't fully thought through the implications but I suspect that this will need to have a different symbol so that you can distinguish them.

For one, how are you going to serialize this? There's no way to get to the name but if you add a field here, then it's a different type than the `$$typeof` represents.

Reply:
Yes agreed. I forgot to mention in the summary but this is part 1. In the second part I'll add the hooks and add warnings to make everything work on the client. This was mostly to get all of the tests to pass with this PR. Btw, should this feature be behind a flag?

Reply:
Another option is to set the Prototype to something else so that we can easily tell on the client. I think that would be sufficient for us?

Reply:
Yea it should be behind a flag. At least in the "isomorphic" and client parts but since that also affects the server (since it depends on isomorphic), you might as well flag it in Flight as well.

We don't use prototypes anywhere else. One reason is that this was faster in benchmarks but another reason is that it works if you have multiple copies of React. That's a little different in this case since the shared state doesn't work properly across multiple copies anyway.